### PR TITLE
flag SignerInformationEntity for deletion

### PR DESCRIPTION
**Pull request by bergmann-dierk**

### CODE REVIEW

File: src/main/java/eu/europa/ec/dgc/gateway/entity/SignerInformationEntity.java [hunk 1]
```diff
@@ -63,13 +66,13 @@
     /**
      * SHA-256 Thumbprint of the certificate (hex encoded).
      */
-    @Column(name = "thumbprint", nullable = false, length = 64, unique = true)
+    @Column(name = "thumbprint", length = 64, unique = true)
```
Comment (f11h) [inline src/main/java/eu/europa/ec/dgc/gateway/entity/SignerInformationEntity.java:69]:
```suggestion
    @Column(name = "thumbprint", nullable = false, length = 64, unique = true)
```

File: src/main/java/eu/europa/ec/dgc/gateway/service/SignerInformationService.java [hunk 2]
```diff
@@ -293,12 +293,16 @@ public void deleteSignerCertificate(
...
+        SignerInformationEntity signerInformationEntity = contentCheckExists(uploadedCertificate);
 
         log.info("Revoking SignerInformation Entity");
 
-        // All checks passed --> Delete from DB
-        signerInformationRepository.deleteByThumbprint(certificateUtils.getCertThumbprint(uploadedCertificate));
+        // All checks passed --> Delete from DB, set fields to null
+        signerInformationEntity.setDeletedAt(ZonedDateTime.now());
+        signerInformationEntity.setRawData(null);
+        signerInformationEntity.setThumbprint(null);
```
Comment (f11h) [inline src/main/java/eu/europa/ec/dgc/gateway/service/SignerInformationService.java:303]:
```suggestion
```

File: src/main/resources/db/changelog/alter-signer-information-for-deletion.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.3.xsd"
+        objectQuotingStrategy="QUOTE_ONLY_RESERVED_WORDS">
+    <changeSet id="alter-signer-information-for-deletion" author="bergmann-dierk">
+        <addColumn tableName="signer_information">
+            <column name="deleted_at" type="DATETIME"/>
+        </addColumn>
+        <dropNotNullConstraint tableName="signer_information" columnName="raw_data"/>
+        <dropNotNullConstraint tableName="signer_information" columnName="thumbprint"/>
```
Comment (f11h) [inline src/main/resources/db/changelog/alter-signer-information-for-deletion.xml:13]:
```suggestion
```

**Review by f11h — CHANGES_REQUESTED** [review]
I would suppose to keep the thumbprint of a certificates - otherwise the MS wouldn't know which certificates was deleted.

**Comment by bergmann-dierk** [comment]
> I would suppose to keep the thumbprint of a certificates - otherwise the MS wouldn't know which certificates was deleted.

Keeping raw_data and Thumbprint but returning raw_data as null.
raw_data is needed to calculate kid

**Pull request merged**